core/src/builder/transaction/challenge.rs

```diff
@@ -9,53 +9,15 @@ use crate::rpc::clementine::{NormalSignatureKind, WatchtowerSignatureKind};
 use bitcoin::{Amount, ScriptBuf, Sequence, TxOut, XOnlyPublicKey};
 use std::sync::Arc;
 
-/// Creates a [`TxHandler`] for the `watchtower_challenge_kickoff_tx`. This transaction can be sent by anyone.
-/// When spent, the outputs of this transaction will reveal the Groth16 proofs with their public inputs for the longest
-/// chain proof, signed by the corresponding watchtowers using WOTS.
-pub fn create_watchtower_challenge_kickoff_txhandler(
-    kickoff_tx_handler: &TxHandler,
-    num_watchtowers: u32,
-    watchtower_xonly_pks: &[XOnlyPublicKey],
-    watchtower_challenge_winternitz_pks: &[Vec<[u8; 20]>],
-    network: bitcoin::Network,
-) -> Result<TxHandler, BridgeError> {
-    let mut builder = TxHandlerBuilder::new().add_input(
-        NormalSignatureKind::WatchtowerChallengeKickoff,
-        kickoff_tx_handler.get_spendable_output(0)?,
-        SpendPath::ScriptSpend(0),
-        DEFAULT_SEQUENCE,
-    );
-
-    for i in 0..num_watchtowers {
-        let winternitz_commit = Arc::new(WinternitzCommit::new(
-            watchtower_challenge_winternitz_pks[i as usize].clone(),
-            watchtower_xonly_pks[i as usize],
-            240,
-        ));
-        builder = builder.add_output(UnspentTxOut::from_scripts(
-            Amount::from_sat(2000), // TODO: Hand calculate this
-            vec![winternitz_commit],
-            None,
-            network,
-        ));
-    }
-
-    Ok(builder
-        .add_output(UnspentTxOut::from_partial(
-            builder::transaction::anchor_output(),
-        ))
-        .finalize())
-}
-
 /// Creates a "simplified "[`TxHandler`] for the `watchtower_challenge_kickoff_tx`. The purpose of the simplification
 /// is that when the verifiers are generating related sighashes, they only need to know the output addresses or the
 /// input UTXOs. They do not need to know output scripts or spendinfos.
-pub fn create_watchtower_challenge_kickoff_txhandler_simplified(
+pub fn create_watchtower_challenge_kickoff_txhandler_from_db(
```

Comment:
```suggestion
pub fn create_watchtower_challenge_kickoff_txhandler(
```
with Option<Arc<WinternitzCommit>>